src/test/java/io/zentity/resolution/AbstractITCase.java

```diff
@@ -6,43 +6,65 @@
 import org.elasticsearch.client.Request;
 import org.elasticsearch.client.Response;
 import org.elasticsearch.client.RestClient;
-import org.elasticsearch.test.ESTestCase;
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
 
+import java.io.File;
 import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
 
-import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.not;
-import static org.junit.Assume.assumeThat;
-
-public abstract class AbstractITCase extends ESTestCase {
-    protected final static int HTTP_TEST_PORT = 9400;
-    protected static RestClient client;
-
-    @BeforeClass
-    public static void startRestClient() throws IOException {
-        client = RestClient.builder(new HttpHost("localhost", HTTP_TEST_PORT)).build();
-        try {
-            Response response = client.performRequest(new Request("GET", "/"));
-            JsonNode json = Json.MAPPER.readTree(response.getEntity().getContent());
-            assertTrue(json.get("tagline").textValue().equals("You Know, for Search"));
-        } catch (IOException e) {
-            // If we have an exception here, let's ignore the test
-            assumeThat("Integration tests are skipped", e.getMessage(), not(containsString("Connection refused")));
-            fail("Something wrong is happening. REST Client seemed to raise an exception.");
-            if (client != null) {
-                client.close();
-                client = null;
-            }
-        }
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeFalse;
+
+public abstract class AbstractITCase {
+
+    public final static String SERVICE_NAME = "es01";
+    public final static int SERVICE_PORT = 9400;
+
+    // Create a singleton docker-compose cluster for all test classes.
+    private static final DockerComposeContainer cluster;
```

Comment:
Instead of booting up this cluster in the static block, what do you think about using either a [`@ClassRule`](https://www.baeldung.com/junit-4-rules#introduction-to-junit-4-rules) or in `@Before / @After` hooks? This will spin up multiple clusters during tests but provides much more isolation between suites.

Reply:
That was my first approach before I settled on testcontainer's [singleton example](https://www.testcontainers.org/test_framework_integration/manual_lifecycle_control/#singleton-containers). I had some challenges with the prior approach. I'm also trying to be cognizant of the 2,000 monthly minutes for GitHub Actions ([here](https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions#about-billing-for-github-actions)); having one cluster will keep the tests expeditious. I made a [similar change](https://github.com/zentity-io/zentity/pull/67/commits/5ca959498ff839d09dfb3b86febf57f9a43312af) in PR #67 to help save on minutes, too.

Reply:
Ah ok, that makes sense. I was confused about the actions minutes as well, but I think the 2,000 mins/ month limit is only for private repositories. 

> GitHub Actions usage is free for both **public repositories** and self-hosted runners. For private repositories, each GitHub account receives a certain amount of free minutes and storage, depending on the product used with the account.

A similar note is on the [overview page](https://github.com/features/actions) too in the "pricing" section.

Reply:
Ah I think you're right. My usage report for GitHub Actions shows 0 minutes. It's only counting the private minutes. Generous!

Reply:
I'll make another attempt at the `@ClassRule` / `@Before` / `@After` approach tonight.